### PR TITLE
TIP-1223: Create query to get IndexableProduct model

### PR DESCRIPTION
**Pull request by nmarniesse**
**Description (for Contributor and Core Developer)**

[TIP-1223](https://akeneo.atlassian.net/browse/TIP-1223)  
- Add `GetIndexableProductInterface`
- Add its implementation. Method `fromProductIdentifier` returns an `IndexableProduct` (or null of product is unknown in DB)
- Update ProductIndexer to use this

No test is done as tests already exist on the indexation, and this will be updated in next card

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Indexer/ProductIndexer.php [hunk 1]
```diff
@@ -23,33 +22,21 @@
...
-    /** @var NormalizerInterface */
-    private $normalizer;
-
+    
     /** @var Client */
     private $productAndProductModelClient;
 
-    /** @var ProductRepositoryInterface */
-    private $productRepository;
+    /** @var \Akeneo\Pim\Enrichment\Bundle\Elasticsearch\GetElasticsearchProductProjectionInterface */
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Indexer/ProductIndexer.php]: Short class name is enough

```
    /** @var GetElasticsearchProductProjectionInterface */
```

Not a big deal

File: tests/back/Integration/IntegrationTestsBundle/Configuration/CatalogInterface.php [hunk 2]
```diff
@@ -14,6 +14,8 @@
 interface CatalogInterface
 {
     /**
+     * @deprecated please do not use it as SQL is not maintainable at all
```
Comment (mmetayer) [inline tests/back/Integration/IntegrationTestsBundle/Configuration/CatalogInterface.php]: :+1:

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/GetElasticsearchProductProjection.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+use Akeneo\Pim\Enrichment\Bundle\Elasticsearch\GetAdditionalPropertiesForProductProjectionInterface;
+use Akeneo\Pim\Enrichment\Bundle\Elasticsearch\GetElasticsearchProductProjectionInterface;
+use Akeneo\Pim\Enrichment\Bundle\Elasticsearch\Model\ElasticsearchProductProjection;
+use Akeneo\Pim\Enrichment\Component\Product\EntityWithFamilyVariant\EntityWithFamilyVariantAttributesProvider;
+use Akeneo\Pim\Enrichment\Component\Product\Exception\ObjectNotFoundException;
+use Akeneo\Pim\Enrichment\Component\Product\Normalizer\Indexing\ProductAndProductModel\ProductModelNormalizer;
+use Akeneo\Pim\Enrichment\Component\Product\Query\GetProductCompletenesses;
+use Akeneo\Pim\Enrichment\Component\Product\Repository\ProductRepositoryInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Webmozart\Assert\Assert;
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/GetElasticsearchProductProjection.php]: ?

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Indexer/ProductIndexer.php (hunk 1 above)
Comment (ahocquard) [reply src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Indexer/ProductIndexer.php]: You're right, I will fix it in next PR

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/GetElasticsearchProductProjection.php (hunk 3 above)
Comment (ahocquard) [reply src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/GetElasticsearchProductProjection.php]: mhhh I will refactor it completly in next PR, not a big deal

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Indexer/ProductIndexer.php [hunk 4]
```diff
@@ -68,30 +55,25 @@ public function indexFromProductIdentifier(string $productIdentifier, array $opt
...
-                continue;
-            }
-
-            $normalizedProduct = $this->normalizer->normalize(
-                $object,
-                ProductModelNormalizer::INDEXING_FORMAT_PRODUCT_AND_MODEL_INDEX
-            );
-            $this->validateObjectNormalization($normalizedProduct);
-            $normalizedProducts[] = $normalizedProduct;
+        if (empty($productIdentifiers)) {
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/Indexer/ProductIndexer.php]: As usual, useless IMHO.

File: tests/back/Pim/Enrichment/Integration/Product/Query/Sql/GetElasticsearchProductProjectionIntegration.php [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+use AkeneoTest\Pim\Enrichment\Integration\Normalizer\NormalizedProductCleaner;
+use PHPUnit\Framework\Assert;
+
+/**
+ * Integration tests to check that the projection of the product is correctly fetched from the database.
+ * A test exists also in EE to check that it fetches the required properties.
+ */
+class GetElasticsearchProductProjectionIntegration extends TestCase
+{
+    protected function setUp(): void
```
Comment (pierallard) [inline tests/back/Pim/Enrichment/Integration/Product/Query/Sql/GetElasticsearchProductProjectionIntegration.php]: useless

**Pull request merged**